identity/aziot-identityd/src/main.rs

```diff
@@ -45,29 +45,43 @@ async fn run() -> Result<(), Box<dyn std::error::Error>> {
 	let server = aziot_identityd::Server::new(settings, authenticator, authorizer)?;
 	let server = std::sync::Arc::new(futures_util::lock::Mutex::new(server));
 	{
-		log::info!("Provisioning starting..");
-
 		let mut server_ = server.lock().await;
-		let device = server_.provision_device().await?;
-		log::info!("Provisioning complete.");
+		let curr_device_info_state;
 
-		let curr_device_info_state = aziot_identityd::settings::DeviceInfo { hub_name: device.iothub_hostname, device_id: device.device_id };
+		if let aziot_identityd::settings::ProvisioningType::None = server_.settings.provisioning.provisioning {
+			log::info!("Skipping provisioning with IoT Hub.");
 
-		if prev_device_info_path.exists() {
-			let prev_device_info_state = aziot_identityd::settings::DeviceInfo::new(&prev_device_info_path)?;
+			curr_device_info_state = aziot_identityd::settings::DeviceInfo {
```

Comment:
We shouldn't do this, otherwise we will return invalid information to callers of `/identities/device`

Reply:
Fixed